File: 2015/08.py
# pylint: disable=missing-module-docstring,missing-function-docstring
# pylint: disable=invalid-name
import sys
import simplejson as json

def part_one(lines):
    """ part one """
    return sum(len(line) - len(eval(f'{line}')) for line in lines)

def part_two(lines):
    """ part two """
    return sum(len(json.dumps(line)) - len(line) for line in lines)

def main():
    """ main """
    lines = []
    for line in sys.stdin:
        line = line.replace('\n', '')
        lines.append(line)

    print('part_one', part_one(lines))
    print('part_two', part_two(lines))


main()

File: 2015/08.js
const readline = require('readline')
const rl = readline.createInterface({ input: process.stdin })

const lines = []

rl.on('line', data => {
  const line = data
  
  lines.push(line)
})

rl.on('close', () => {
  console.log('partOne', partOne(lines))
  console.log('partTwo', partTwo(lines))
})

function partOne (lines) {
  for (const line of lines) {
    console.log(line, JSON.stringify(line))
  }
  return 'todo'
}

function partTwo (lines) {
  return 'todo'
}
